### DISCUSSION TITLE
PIP error

### DISCUSSION THREAD

**Discussion by Cossackx** [khoj-ai/khoj#194] [Q&A]
Any idea how to fix this?

  Using cached khoj_assistant-0.1.2-py3-none-any.whl (142 kB)
  Using cached khoj_assistant-0.1.0-py3-none-any.whl (51 kB)
ERROR: Cannot install khoj-assistant==0.1.0, khoj-assistant==0.1.10, khoj-assistant==0.1.2, khoj-assistant==0.1.3, khoj-assistant==0.1.4, khoj-assistant==0.1.5, khoj-assistant==0.1.6, khoj-assistant==0.1.7, khoj-assistant==0.1.8, khoj-assistant==0.1.9, khoj-assistant==0.2.0, khoj-assistant==0.2.1 and khoj-assistant==0.2.5 because these package versions have conflicting dependencies.

The conflict is caused by:
    khoj-assistant 0.2.5 depends on torch==1.13.1
    khoj-assistant 0.2.1 depends on torch==1.12.1
    khoj-assistant 0.2.0 depends on torch==1.12.1
    khoj-assistant 0.1.10 depends on torch==1.12.1
    khoj-assistant 0.1.9 depends on torch==1.12.1
    khoj-assistant 0.1.8 depends on torch==1.12.1
    khoj-assistant 0.1.7 depends on torch==1.12.1
    khoj-assistant 0.1.6 depends on torch==1.11.0
    khoj-assistant 0.1.5 depends on torch==1.11.0
    khoj-assistant 0.1.4 depends on torch==1.11.0
    khoj-assistant 0.1.3 depends on torch==1.11.0
    khoj-assistant 0.1.2 depends on torch==1.11.0
    khoj-assistant 0.1.0 depends on torch==1.11.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

**Comment by debanjum** [comment]
Hey @Cossackx, what is your python version? (You can check that with `python -V`). 

This could be because the python version you're using isn't supported by Khoj. 

Khoj currently work with python 3.8 to 3.10. (As python 3.11 isn't supported by pytorch, a dependency for Khoj)

**Comment by Cossackx** [comment]
@debanjum That's it! I'm on 3.11. Thanks for the sanity check!

**Comment by Cossackx** [comment]
@debanjum, maybe you can assist again...

When I do this step...
1. Setup Backend
Open terminal/cmd and run below command to install and start the khoj backend

pip install khoj-assistant && khoj --no-gui

I get...
10\site-packages (from anyio<5,>=3.4.0->starlette==0.19.1->fastapi==0.77.1->khoj-assistant) (1.3.0)
'khoj' is not recognized as an internal or external command,
operable program or batch file.

Thank you very much!

**Comment by debanjum** [comment]
I'm guessing this is because the Python scripts directory is not added to your [PATH](https://www.wikihow.com/Change-the-PATH-Environment-Variable-on-Windows) (where installed scripts/commands are searched for)

Assuming you're using python 3.10 (and on windows), to run the khoj script from its installed location try run this from your terminal: `C:\Users\%USER%\AppData\Local\Programs\Python\Python310\Scripts\khoj.exe --no-gui`

**Comment by debanjum** [comment]
Converted this to a discussion as it doesn't seem like a bug with khoj. 
Feel free to respond here, if you're still having issues installing khoj!
